src/mlpack/tests/main_tests/range_search_utils.hpp

```diff
@@ -0,0 +1,103 @@
+/**
+ * @file hmm_test_utils.hpp
+ * @author Niteya Shah
+ *
+ * Helper Functions used in the execution of the CLI Range Search Test
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_TESTS_MAIN_TESTS_RANGE_SEARCH_TEST_UTILS_HPP
+#define MLPACK_TESTS_MAIN_TESTS_RANGE_SEARCH_TEST_UTILS_HPP
+
+#include <boost/test/unit_test.hpp>
+#include <mlpack/methods/range_search/rs_model.hpp>
+#include <mlpack/core.hpp>
+#include <mlpack/core/util/mlpack_main.hpp>
+/*
+* Convert a Model to String by calling the RSModel serialize function of the
+* boost library and return the Model in String Form
+* @param model - RSModel to be converted to string
+*/
+inline std::string ModelToString(RSModel* model)
+{
+  std::ostringstream oss;
+  boost::archive::text_oarchive oa(oss);
+  oa << model;
+  return oss.str();
+}
+
+/*
+* Check for 2 matrices of type vector<vector<double>> to ensure that their
+* values dont differ by more than tolerance , default is 0.001%
+* @param vec1 - vector 1 to be checked
+* @param vec2 - vector 2 to be checked
+* @param tolerance - difference in values in allowed
+*/
+inline void CheckMatrices(std::vector<std::vector<double>> vec1,
+                          std::vector<std::vector<double>> vec2,
```

Comment:
Using const reference throws an error because we have to sort the vectors later to compare them , so I made them reference the original , but they are not const .